### ISSUE TITLE
boinc-client daemon does not start automatically and hence cannot connect

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1133]
**Reported by shirish on 20 Aug 41935076 09:33 UTC**
Hi all, 

I purged and removed the old boinc-manager removing also the
/var/lib/boinc-client directory due to some graphical issues I was
facing then. 

Then I installed boinc-client and boinc-manager afresh and
rebooted the system. The rebooting was doing in the hopes that the
boinc-client daemon would start automatically but it seems it doesn't.

According to http://wiki.debian.org/BOINC which has a simple two-liner
for users, it says and I quote

"BOINC comes as an official part of Debian. Just install the boinc
package. Then start the boincmgr application to join the project of
your liking."

But if I try to use boincmgr and try to attach the project, two things happen.

a. On the GUI I get after a minute or two :-

The BOINC client has exited unexpectedly 3 times within the last 2
minutes. Would you like to restart it again?

b. On the CLI I get this :-

```
$ boincmgr
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
```

Now I tried to figure out if boinc daemon is started or not and by
default it seems it is not started.

```
$ boinccmd --get_state
can't connect to local host
```

```
$ sudo /etc/init.d/boinc-client status
Status of BOINC core client: stopped.
```

I could simply just to $sudo /etc/init.d/boinc-client start but this
should not be needed, right ?

c. To top on things, if I try to use boinc-manager > Advanced > Select
Computer it asks for Host Name and password. In my case the Host Name
would be localhost.But the moment I put localhost therein I get a big
password which I don't know from where its coming.

d. I am confused, another part of the issue wasn't/isn't Multiarch
supposed to get rid of ia32-libs ? http://wiki.debian.org/Multiarch

Looking forward to know where am I going wrong.

-- System Information:
Debian Release: wheezy/sid
 APT prefers unstable
 APT policy: (500, 'unstable'), (500, 'testing'), (500, 'stable'),
(1, 'experimental')
Architecture: amd64 (x86_64)

Kernel: Linux 3.1.0-1-amd64 (SMP w/2 CPU cores)
Locale: LANG=en_IN, LC_CTYPE=en_IN (charmap=UTF-8)
Shell: /bin/sh linked to /bin/dash

Versions of packages boinc-manager depends on:
ii  libc6           2.13-21
ii  libgcc1         1:4.6.2-5
ii  libglib2.0-0    2.30.2-4
ii  libgtk2.0-0     2.24.8-2
ii  libnotify4      0.7.4-1
ii  libsqlite3-0    3.7.9-2
ii  libstdc++6      4.6.2-5
ii  libwxbase2.8-0  2.8.12.1-3
ii  libwxgtk2.8-0   2.8.12.1-3

Versions of packages boinc-manager recommends:
ii  boinc-client   7.0.2+dfsg-1
ii  ia32-libs-gtk  20111001

Versions of packages boinc-manager suggests:
ii  libgl1-mesa-glx  7.11.1-1
ii  libxt6           1:1.1.1-2

-- no debconf information

This is already filed in debian at http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651303 

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1160

**Comment by romw** [comment]
**Commented by shirish on 25 Jun 41935125 18:19 UTC**
I purged and removed the old boinc-manager removing also the /var/lib/boinc-client directory due to some graphical issues I was facing then. Then installed boinc-client and boinc-manager afresh and rebooted the system. The rebooting was doing in the hopes that the boinc-client daemon would start automatically but it seems it doesn't.

According to http://wiki.debian.org/BOINC which has a simple two-liner for users, it says and I quote

"BOINC comes as an official part of Debian. Just install the boinc package. Then start the boincmgr application to join the project of your liking."

But if I try to use boincmgr and try to attach the project, two things happen.

 a. On the GUI I get after a minute or two :-

The BOINC client has exited unexpectedly 3 times within the last 2 minutes. Would you like to restart it again?

 a. On the CLI I get this :-

```
$ boincmgr
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager)
failed with error 2!
```
Now I tried to figure out if boinc daemon is started or not and by default it seems it is not started.

```
$ boinccmd --get_state
can't connect to local host
```
```
$ sudo /etc/init.d/boinc-client status
Status of BOINC core client: stopped.
```
I could simply just to $sudo /etc/init.d/boinc-client start but this should not be needed, right ?

 a. To top on things, if I try to use boinc-manager > Advanced > Select

Computer it asks for Host Name and password. In my case the Host Name would be localhost.But the moment I put localhost therein I get a big password which I don't know from where its coming.

 a. I am confused, another part of the issue wasn't/isn't Multiarch

supposed to get rid of ia32-libs ? http://wiki.debian.org/Multiarch

Looking forward to know where am I going wrong.

-- System Information:

Debian Release: wheezy/sid

APT prefers unstable APT policy: (500, 'unstable'), (500, 'testing'), (500, 'stable'), (1, 'experimental')

Architecture: amd64 (x86_64)

Kernel: Linux 3.1.0-1-amd64 (SMP w/2 CPU cores) 

Locale: LANG=en_IN, LC_CTYPE=en_IN (charmap=UTF-8) 

Shell: /bin/sh linked to /bin/dash

Versions of packages boinc-manager depends on:

ii  libc6           2.13-21

ii  libgcc1         1:4.6.2-5

ii  libglib2.0-0    2.30.2-4

ii  libgtk2.0-0     2.24.8-2

ii  libnotify4      0.7.4-1

ii  libsqlite3-0    3.7.9-2

ii  libstdc++6      4.6.2-5

ii  libwxbase2.8-0  2.8.12.1-3

ii  libwxgtk2.8-0   2.8.12.1-3

Versions of packages boinc-manager recommends:

ii  boinc-client   7.0.2+dfsg-1

ii  ia32-libs-gtk  20111001

Versions of packages boinc-manager suggests:

ii  libgl1-mesa-glx  7.11.1-1 ii  libxt6           1:1.1.1-2

-- no debconf information

**Comment by romw** [comment]
**Commented by shirish on 10 Aug 41935163 17:37 UTC**
Already filed as http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651303 in Debian.

**Comment by romw** [comment]
**Commented by romw on 4 Sep 41935180 03:16 UTC**
Under that configuration, boinc/boincmgr/boinccmd all need to be in /home/shirish/boinc and the CWD needs to be set to /home/shirish/boinc before launching any of them.

**Comment by romw** [comment]
**Commented by shirish on 16 Nov 41935183 07:56 UTC**
what's CWD ?

**Comment by romw** [comment]
**Commented by romw on 14 Feb 41935185 07:15 UTC**
Current Working Directory.

**Comment by romw** [comment]
**Commented by romw on 21 Jul 41935187 22:26 UTC**
IIRC, error code 2 on *nix systems is file not found.

**Comment by romw** [comment]
**Commented by shirish on 22 Nov 41935205 13:46 UTC**
ah ok, just ran it in verbose mode, got some more output this time :-

```
~$ boincmgr --verbose
12:41:22  IST: looking for catalog 'BOINC-Manager' in path '/home/shirish/locale/.UTF-8/LC_MESSAGES:/home/shirish/locale/:/home/shirish/locale/.UTF-8:locale/.UTF-8/LC_MESSAGES:locale/:locale/.UTF-8:/usr/share/locale/.UTF-8/LC_MESSAGES:/usr/share/locale/.UTF-8/LC_MESSAGES:/usr/share/locale/:/usr/share/locale/.UTF-8:/home/shirish/locale//LC_MESSAGES:/home/shirish/locale/:/home/shirish/locale/:locale//LC_MESSAGES:locale/:locale/:/usr/share/locale//LC_MESSAGES:/usr/share/locale//LC_MESSAGES:/usr/share/locale/:/usr/share/locale/'.
12:41:22  IST: catalog file for domain 'BOINC-Manager' not found.
12:41:22  IST: looking for catalog 'BOINC-Client' in path '/home/shirish/locale/.UTF-8/LC_MESSAGES:/home/shirish/locale/:/home/shirish/locale/.UTF-8:locale/.UTF-8/LC_MESSAGES:locale/:locale/.UTF-8:/usr/share/locale/.UTF-8/LC_MESSAGES:/usr/share/locale/.UTF-8/LC_MESSAGES:/usr/share/locale/:/usr/share/locale/.UTF-8:/home/shirish/locale//LC_MESSAGES:/home/shirish/locale/:/home/shirish/locale/:locale//LC_MESSAGES:locale/:locale/:/usr/share/locale//LC_MESSAGES:/usr/share/locale//LC_MESSAGES:/usr/share/locale/:/usr/share/locale/'.
12:41:22  IST: catalog file for domain 'BOINC-Client' not found.
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager) failed with error 2!
execvp(/home/shirish/boinc, --redirectio, --launched_by_manager) failed with error 2!
```

So something called catalog file is not being found, any idea what this catalog file is ?

**Comment by romw** [comment]
**Commented by romw on 11 Jul 41935211 16:04 UTC**
The catalog file is where translated text is stored. BOINC-Manager.mo and BOINC-Client.mo.

Not part of the problem the bug report is about.

**Comment by romw** [comment]
**Commented by shirish on 6 Dec 41935218 01:09 UTC**
so that failed with error 2 stuff is about the .mo files being not found or something else ?

**Comment by romw** [comment]
**Commented by romw on 23 May 41935395 01:58 UTC**
boincmgr is not able to launch boinc, it is not able to find boinc.

**Comment by ChristianBeer** [comment]
Should be fixed in current 7.6.33 and also in Debian.
